Jenkinsfile

```diff
@@ -134,5 +134,35 @@ pipeline {
                 }
             }
         }
+        stage('Publish image to DockerHub') {
+            agent {
+                label 'team-local && linux'
+            }
+            when {
+                buildingTag()
+            }
+            environment {
+                DOCKERHUB_CREDS=credentials('jenkins-docker-hub-credentials')
+            }
+            steps {
+                dir('src/github.com/docker/app') {
+                    checkout scm
+                    dir("bin") {
+                        unstash "binaries"
+                    }
+                    script {
+                        docker.withRegistry('https://registry.hub.docker.com', 'jenkins-docker-hub-credentials') {
```

Comment:
Are this line and the `DOCKERHUB_CREDS=credentials('jenkins-docker-hub-credentials')` redundant with each other? I'd expect `docker.WithRegistry` to set some environment variables with the credentials but maybe it just does the `docker login` and nothing else? (docs aren't very clear)

Reply:
The user and passwd are needed inside `update-dockerhub-repo.sh` as environment variables.
if `docker.withRegistry` exposes them as env vars that would be perfect